test/test_viz.py

```diff
@@ -0,0 +1,48 @@
+import jax.numpy as jnp
+import pygame
+import pygame.locals
+
+from cotix._bodies import Ball
+from cotix._visualizer_tools import BallDrawer, Visualizer
+
+
+def test_viz():
+    framerate = 60
+
+    viz = Visualizer(100, 6, 6, name="Test Viz")
+
+    b = Ball.make_default()
+
+    b = b.set_position(b.position + jnp.array([3.0, 3.0]))
+
+    print(b.position)
+
+    b_drawer = BallDrawer(b, pygame.Color(255, 0, 0))
+
+    run = True
+    framecount = 0
+    inc = jnp.array([0.1, 0.1])
+
+    while run:
+        # Iterating over all the events received from
+        # pygame.event.get()
+        for event in viz.get_events():
+            # If the type of the event is quit
+            # then setting the run variable to false
+            if event.type == pygame.locals.QUIT:
+                run = False
+        # Draws the surface object to the screen.
+
+        viz.draw([b_drawer])
+
+        pygame.time.Clock().tick(framerate)
+
+        framecount += 1
+
+        # update the ball
+        b = b.set_position(b.position + inc)
+        # update the drawer with the ball
+        b_drawer.object = b
```

Comment:
Also you are not allowed to do b_drawer.object = b

Reply:
I mean, since it is not JAX, you are allowed to do so, but since we are trying to make your simulator in the same style as JAX, it should adhere

BTW, one cool test that it works under jit would be that you use ```python jax.debug.callback``` to do any pygame-related stuff, and then just put everything you do into jitted region. Then you will be able to see whether your code actually adheres to the standard (since otherwise it won't compile), and it is also more convenient for us, since then your simulator will be less restrictive in terms of control flow: now we could do calls from inside jit, if we want to, and allow the user decide when to do visualizations. I'd like to see this implemented, but it might be a bit too complex

Reply:
So instead of doing 

       b_drawer.object = b

We should create the b_drawer from scratch with the new b info, right?

Reply:
either this, or better you could use ```eqx.tree_at(lambda drawer: drawer.object, b_drawer, replace=b)```